### PR TITLE
Added Windows Functionality to TerminalOperations

### PR DESCRIPTION
**Pull request by yaraalkurdi**

### CODE REVIEW

File: src/main/java/use_case/TerminalOperations/TerminalOperations.java [hunk 1]
```diff
@@ -36,12 +38,16 @@ public void executeCommand(String command, CommandCallback callback) {
 
     private void clearTerminal(CommandCallback callback) throws IOException, InterruptedException {
         callback.onOutput(""); // Clear the terminal UI
-        executeProcess(new String[]{"bash", "-c", "clear"}, callback);
+        if (isWindows) {
```
Comment (dariosanp) [inline src/main/java/use_case/TerminalOperations/TerminalOperations.java:41]: Great solution; adding a switch was definitely the way to go!

File: src/main/java/use_case/TerminalOperations/TerminalOperations.java
Comment (dariosanp) [inline src/main/java/use_case/TerminalOperations/TerminalOperations.java:1]: Great implementation of multi-OS adaptor! Approved

**Review by dariosanp — APPROVED** [review]

File: src/main/java/use_case/TerminalOperations/TerminalOperations.java [hunk 2]
```diff
@@ -8,9 +8,11 @@
```
Comment (AriGood) [inline src/main/java/use_case/TerminalOperations/TerminalOperations.java:8]: error handling is not done at where are the errors that are getting thrown being dealt with?

File: src/main/java/use_case/TerminalOperations/TerminalOperations.java [hunk 3]
```diff
@@ -36,12 +38,16 @@ public void executeCommand(String command, CommandCallback callback) {
```
Comment (AriGood) [inline src/main/java/use_case/TerminalOperations/TerminalOperations.java:38]: add a docstring here

**Review by AriGood — APPROVED** [review]

**Pull request merged**